src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -765,23 +751,15 @@ private static string RetrieveProcessUserName(Process process)
                 }
 
                 var tokenUser = Marshal.PtrToStructure<Win32Native.TOKEN_USER>(tokenUserInfo);
-
-                // Max username is defined as UNLEN = 256 in lmcons.h
-                // Max domainname is defined as DNLEN = CNLEN = 15 in lmcons.h
-                // The buffer length must be +1, last position is for a null string terminator.
-                int userNameLength = 257;
-                int domainNameLength = 16;
-                Span<char> userNameStr = stackalloc char[userNameLength];
-                Span<char> domainNameStr = stackalloc char[domainNameLength];
-                Win32Native.SID_NAME_USE accountType;
-
-                // userNameLength and domainNameLength will be set to actual lengths.
-                if (!Win32Native.LookupAccountSid(null, tokenUser.User.Sid, userNameStr, ref userNameLength, domainNameStr, ref domainNameLength, out accountType))
+                SecurityIdentifier sid = new SecurityIdentifier(tokenUser.User.Sid);
+                try
+                {
+                    return sid.Translate(typeof(System.Security.Principal.NTAccount)).Value;
+                }
+                catch (IdentityNotMappedException)
                 {
                     return null;
                 }
```

Comment:
It seems now this try/catch can be merged to the outer try/catch, like this:

```
    var tokenUser = Marshal.PtrToStructure<Win32Native.TOKEN_USER>(tokenUserInfo);
    SecurityIdentifier sid = new SecurityIdentifier(tokenUser.User.Sid);
    userName = sid.Translate(typeof(System.Security.Principal.NTAccount)).Value;
}
catch (IdentityNotMappedException)
{
    // SID cannot be mapped to a user
}
catch (NotSupportedException)
{
    // The Process not started yet, or it's a process from a remote machine.
}
...
```

Reply:
It's already having those outer catch blocks so I think it's better to follow the style. Also, you can always add comment in the catch block to indicate where the exception comes from.

Reply:
> It's already having those outer catch blocks so I think it's better to follow the style

I don't really agree with that logic but so be it.